internal/semaphore.go

```diff
@@ -1,57 +0,0 @@
-// Copyright (C) MongoDB, Inc. 2017-present.
-//
-// Licensed under the Apache License, Version 2.0 (the "License"); you may
-// not use this file except in compliance with the License. You may obtain
-// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
-
-package internal
-
-import (
-	"context"
-	"errors"
-)
-
-// NewSemaphore creates a new semaphore.
-func NewSemaphore(slots uint64) *Semaphore {
-	ch := make(chan struct{}, slots)
-	for i := uint64(0); i < slots; i++ {
-		ch <- struct{}{}
-	}
-
-	return &Semaphore{
-		permits: ch,
-	}
-}
-
-// Semaphore is a synchronization primitive that controls access
-// to a common resource.
-type Semaphore struct {
```

Comment:
Was this internal semaphore type just never used?

Reply:
The `internal.Semaphore` type was added at https://github.com/mongodb/mongo-go-driver/commit/7a3bfb5daa304f3099be4511658adba0b495a3dd and used in `server/pool.go`. It was no longer used after `mongo/private/conn/pool.go` was replaced by a complete refactor in `core/connection/pool.go` (used `"golang.org/x/sync/semaphore"` instead) at https://github.com/mongodb/mongo-go-driver/commit/1fa4f118ac3bf15efdf1ae9b2437df03bc078435.